Classes/Service/ListService.php

```diff
@@ -56,6 +57,15 @@ public static function resolveListitems(array &$row, $field = 'tx_listelements_l
             ->execute()
             ->fetchAll();
 
+        foreach ($row[$returnAs] as $key => $specificRow) {
+            BackendUtility::workspaceOL('tx_listelements_item', $specificRow);
+            if ($specificRow !== false) {
```

Comment:
@dgorges 
i think this should be
`if ($specificRow !== false && !VersionState::cast($row['t3ver_state'] ?? 0)->equals(VersionState::DELETE_PLACEHOLDER)) {`
for deleted workspace rows
(with
`use TYPO3\CMS\Core\Versioning\VersionState;`
)
can you test this with deleted workspace row

Reply:
@achimfritz thank you. It works. I just included your code.